node/src/main/scala/org/bitcoins/node/networking/peer/DataMessageHandler.scala

```diff
@@ -94,6 +95,12 @@ case class DataMessageHandler(
             }
           }
           newChainApi <- chainApi.processFilter(filter)
+          _ <- Future {
```

Comment:
So is this `Future` needed because the signature is

```scala
type OnCompactFilterReceived = (DoubleSha256Digest, GolombFilter) => Unit
```

rather than  returning a `Future[Unit]`? Are you worried that the callback is going to take a long time to process? It seems like a valid concern to me.